File: Source/OldCode/Archive/2005 TVA Code Library/Version/4.0/Source/TVA.PhasorProtocols/IHeaderCell.cs
//*******************************************************************************************************
//  IHeaderCell.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R. Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  04/16/2005 - James R. Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System;

namespace TVA.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of any kind of <see cref="IHeaderFrame"/> cell.
    /// </summary>
    public interface IHeaderCell : IChannelCell
    {
        /// <summary>
        /// Gets a reference to the parent <see cref="IHeaderFrame"/> for this <see cref="IHeaderCell"/>.
        /// </summary>
        new IHeaderFrame Parent { get; set; }

        /// <summary>
        /// Gets or sets ASCII character as a <see cref="Byte"/> that represents this <see cref="IHeaderCell"/>.
        /// </summary>
        byte Character { get; set; }
    }
}

File: Source/OldCode/Archive/2006 Phasor Protocols/PhasorProtocols/IHeaderCell.vb
'*******************************************************************************************************
'  IHeaderCell.vb - Header cell interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  04/16/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

''' <summary>This interface represents the protocol independent representation of an element of header data.</summary>
<CLSCompliant(False)> _
Public Interface IHeaderCell

    Inherits IChannelCell

    Shadows ReadOnly Property Parent() As IHeaderFrame

    Property Character() As Byte

End Interface